vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/DRDAQueryRequest.java

```diff
@@ -50,14 +50,15 @@ public DRDAQueryRequest(ByteBuf buffer) {
      * @return True if the SQL is a query (i.e. SELECT) and false otherwise
      */
     public static boolean isQuery(String sql) {
-        sql = sql.trim();
-        return sql.startsWith("SELECT") 
-                || sql.startsWith("select")
-                || sql.startsWith("WITH")
-                || sql.startsWith("with");
+    	if (sql != null) {
```

Comment:
I think if a null is passed in we should blow up with an IllegalArgumentException instead of returning false (pretending it's an update)

Reply:
Done.  Thank you!